src/Carting.Api/Controllers/V1/CartingController.cs

```diff
@@ -0,0 +1,138 @@
+﻿using Carting.Api.Mappers.V1;
+using Carting.Api.Requests.V1;
+using Carting.Core.Services;
+using Carting.DataAccess.Exceptions;
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.ModelBinding;
+
+namespace Carting.Api.Controllers.V1;
+
+[ApiController]
+[ApiVersion("1.0")]
+[ApiExplorerSettings(GroupName = "v1")]
+[Route("api/v{version:apiVersion}")]
+public class CartingController : ControllerBase
+{
+    private readonly ICartingService cartingService;
+
+    public CartingController(ICartingService cartingService)
+    {
+        this.cartingService = cartingService;
+    }
+
+    /// <summary>
+    /// Returns cart items for the provided cartId.
+    /// </summary>
+    /// <param name="cartId">Identifier of the cart.</param>
+    /// <returns>Cart items for the provided cartId.</returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     GET /cart/1
+    ///
+    /// </remarks>
+    /// <response code="200">Cart items returned successfully</response>
+    /// <response code="500">Internal server error occured</response>
+    [HttpGet]
+    [Route("cart/{cartId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status500InternalServerError)]
+    public IActionResult Get(string cartId)
+    {
+        try
+        {
+            var result = cartingService.GetCartItems(cartId);
+
+            return Ok(Mapper.Map(cartId, result));
+        }
+        catch (Exception ex)
+        {
+            return StatusCode(500, ex.Message);
+        }
+    }
+
+    /// <summary>
+    /// Creates a cart item in the cart for the provided cartId.
+    /// </summary>
+    /// <param name="cartId">Identifier of the cart.</param>
+    /// <param name="request">Cart item request.</param>
+    /// <returns></returns>
+    /// <remarks>
+    /// Sample request:
+    ///
+    ///     POST /Cart/1
+    ///     {
+    ///        "name": "string",
+    ///        "image": {
+    ///             "url": "string",
+    ///             "alt": "string",
+    ///        },
+    ///        "price": 0,
+    ///        "quantity": 0
+    ///     }
+    ///
+    /// </remarks>
+    /// <response code="200">Cart items returned successfully</response>
+    /// <response code="400">Bad request</response>
+    /// <response code="500">Internal server error occured</response>
+    [HttpPost]
+    [Route("cart/{cartId}")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    [ProducesResponseType(StatusCodes.Status400BadRequest)]
```

Comment:
Could you show me where this endpoint returns http status 400

Reply:
Could you show show me how `If there was no cart for specified key – creates it` requirement is implemented

Reply:
> Could you show show me how `If there was no cart for specified key – creates it` requirement is implemented

The CartId is just a property for a CartItem, so the CartId parameter will be "assigned" to a CartItem.